src/carlib/family_car.rs

```diff
@@ -1,44 +1,36 @@
 use crate::carlib::car::Car;
-use crate::carlib::car_concept::CarConcept;
-use crate::carlib::car_impl::CarImpl;
 
 // Let's try a different approach with the family car.
+const ACCELERATION: f64 = 0.05; // Quite slow.
 pub struct Familycar {
-    pub imp: CarImpl,
+    //don't need acceleration to be a variable, because it never changes
+    pub name: String,
+    pub speed: f64,
 }
 
 impl Familycar {
-    const ACCELERATION: f64 = 0.05; // Quite slow.
-
-    pub fn new() -> Familycar {
-        let mut obj = Familycar {
-            imp: CarImpl::new(),
-        };
-        obj.imp.acceleration = Familycar::ACCELERATION;
-        obj
+    pub fn new(name: Option<String>) -> Self {
+        Self {
+            name: name.unwrap_or("<unnamed>".to_string()),
+            speed: 0.0,
+        }
     }
 }
 
 // Here we implement the Car trait for the Familycar.
 // This time we use the `delegate!` macro.
 // With this macro we can get rid of quite some boilerplate code.
 impl Car for Familycar {
-    delegate::delegate! {
-        to self.imp {
-            fn accelerate(&mut self, duration: f64);
-            fn brake(&mut self, force: f64);
-            fn get_speed(&self) -> f64;
-            fn get_name(&self) -> &str;
-        }
+    fn get_speed(&self) -> f64 {
+        self.speed
     }
-}
-
-// Read the comment in src/carlib/sports_car.rs for an explanation.
-impl CarConcept for Familycar {
-    fn as_car(&self) -> &dyn Car {
-        &self.imp
+    fn set_speed(&mut self, new_speed: f64) {
+        self.speed = new_speed;
+    }
+    fn get_acceleration(&self) -> f64 {
+        ACCELERATION
     }
-    fn as_mut_car(&mut self) -> &mut dyn Car {
-        &mut self.imp
+    fn get_name(&self) -> &str {
+        &self.name
```

Comment:
I notice that these functions are implemented here and they are implemented again in `Sportscar` in exactly the same way 🐸. In this case they are trivial but having to implement the same 4 functions over and over again is not very DRY, they are essentially duplicates and I would rather like to inherit them from the `Car`.

Reply:
Yes, but for simple getter/setter functions, you can use some simple macro to implement that for you.